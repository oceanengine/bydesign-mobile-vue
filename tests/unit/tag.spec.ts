import { later, mount } from '../index';
import { Tag as NativeTag } from '@src';
const Tag = NativeTag as any;
later;
describe('Tag', () => {
  test('click event', () => {
    const click = jest.fn();
    const wrapper = mount(Tag, {
      context: {
        on: {
          click,
        },
      },
    });

    wrapper.trigger('click');
    expect(click).toHaveBeenCalledTimes(1);
  });
  test('custom style', () => {
    const wrapper = mount(Tag, {
      propsData: {
        type: 'primary',
        plain: true,
        round: true,
        size: 'large',
        color: '#666666',
        textColor: 'red',
      },
      slots: {
        default: '标签',
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
