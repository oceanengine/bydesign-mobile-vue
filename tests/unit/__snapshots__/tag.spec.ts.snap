// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag custom style 1`] = `<span class="byted-tag byted-tag--primary byted-tag--large byted-tag--plain byted-tag--round byted-hairline--surround" style="color: red;">标签</span>`;
